<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>304</width>
    <height>388</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLineEdit" name="txtName"/>
     </item>
     <item>
      <widget class="QComboBox" name="txtIPs"/>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLineEdit" name="txtIP"/>
     </item>
     <item>
      <widget class="QSpinBox" name="numPort">
       <property name="maximum">
        <number>100000</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnConnect">
       <property name="text">
        <string>Connect</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLineEdit" name="txtAksName"/>
     </item>
     <item>
      <widget class="QPushButton" name="btnAsk">
       <property name="text">
        <string>ASK</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLineEdit" name="txtPeerIP_Remote"/>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="numPeerPort_Remote">
       <property name="maximum">
        <number>100000</number>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="btnHole">
       <property name="text">
        <string>Hole</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLineEdit" name="txtPeerIP_Local"/>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="numPeerPort_Local">
       <property name="maximum">
        <number>100000</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLineEdit" name="txtDataGram"/>
     </item>
     <item>
      <widget class="QPushButton" name="btnSendDatagram">
       <property name="text">
        <string>SendDatagram</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QPlainTextEdit" name="txtLog"/>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
